app/src/main/java/com/codesoom/assignment/dto/ProductNotFoundException.java

```diff
@@ -0,0 +1,23 @@
+package com.codesoom.assignment.dto;
+
+public class ProductNotFoundException extends RuntimeException{
+    public ProductNotFoundException() {
+        super();
+    }
+
+    public ProductNotFoundException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    public ProductNotFoundException(String message) {
+        super(message);
+    }
+
+    public ProductNotFoundException(Long id) {
+        super("id " + id + "를 가지는 Product가 없습니다.");
+    }
```

Comment:
흔한 실수입니다. 두 가지 점을 생각해 볼 수 있어요.

- 에러 메시지는 "없습니다"인데, 예외 클래스 이름은 "NotFound"(찾지 못함) 입니다. 둘은 다릅니다.
- 데이터를 조회하지 못했다고 해서 정말로 그 상품이 없다고 보장할 수 없습니다. 가장 확실한 사실은 찾지 못했다는 것 뿐입니다.

```suggestion
    public ProductNotFoundException(Long id) {
        super("id " + id + "를 가지는 Product를 찾지 못했습니다.");
    }
```

Reply:
찾지 못했다는 말이 더 정확한 말이 겠네요.